src/dbSta/src/dbNetwork.cc

```diff
@@ -2562,4 +2607,159 @@ dbNetworkObserver::~dbNetworkObserver()
   }
 }
 
+/*
+  Hierarchical network api connections
+ */
+
+dbModule* dbNetwork::getParentModule(dbNet* net)
+{
+  if (hasHierarchy()) {
+    // get sink driver instance and return its parent
+    int drivingITerm = net->getDrivingITerm();
+    if (drivingITerm != 0 && drivingITerm != -1) {
+      dbITerm* iterm = block_->getITerm(drivingITerm);
+      dbModNet* modnet = iterm->getModNet();
+      if (modnet != nullptr) {
+        return modnet->getParent();
+      }
+    }
+    // default to top module
+    return block_->getTopModule();
+  }
+  return nullptr;
+}
+
+void dbNetwork::getInstanceTree(dbModule* start_module,
+                                std::vector<dbModule*>& instance_tree)
+{
+  dbModule* top_module = block_->getTopModule();
+  dbModule* cur_module = start_module;
+  while (cur_module) {
+    instance_tree.push_back(cur_module);
+    if (cur_module == top_module)
+      return;
+    cur_module = start_module->getModInst()->getParent();
+  }
+}
+
+dbModule* dbNetwork::findHighestCommonModule(std::vector<dbModule*>& itree1,
+                                             std::vector<dbModule*>& itree2)
+{
+  // order lowest to highest (push_back in getInstanceTree)
+  for (auto i : itree1) {
+    for (auto j : itree2) {
```

Comment:
Why not start from the top down and find the first divergence?  It would be simpler/faster.